src/Chainweb/Chainweb.hs

```diff
@@ -646,43 +647,64 @@ runChainweb cw = do
 
     -- P2P Server
 
-    serverSettings :: Settings
-    serverSettings = setOnException
-        (\r e -> when (shouldDisplayException e) (logg Warn $ loggServerError r e))
-        $ peerServerSettings (_peerResPeer $ _chainwebPeer cw)
-      where
-        shouldDisplayException e
-            | Just InsecureConnectionDenied <- fromException e = False
-            | otherwise = defaultShouldDisplayException e
+    serverSettings :: IORef Int -> Settings
+    serverSettings closedConnectionsRef = setOnException
+        (\r e -> if
+            | Just InsecureConnectionDenied <- fromException e ->
+                return ()
+            | Just ClientClosedConnectionPrematurely <- fromException e ->
+                atomicModifyIORef' closedConnectionsRef (\n -> (n + 1, ()))
```

Comment:
It might be overkill here, but you could consider using https://github.com/kadena-io/chainweb-node/blob/2.16/src/Chainweb/Counter.hs. One (small) advantage would be the that log message format would be consistent with other counters.